src/ray/common/memory_monitor.cc

```diff
@@ -39,7 +44,8 @@ MemoryMonitor::MemoryMonitor(instrumented_io_context &io_service,
         [this] {
           auto [used_memory_bytes, total_memory_bytes] = GetMemoryBytes();
           MemorySnapshot system_memory;
-          system_memory.used_bytes = used_memory_bytes;
+          system_memory.heap_used_bytes = used_memory_bytes;
+          system_memory.object_store_used_bytes = object_store_memory_usage_fetcher_();
```

Comment:
Why wasn't this necessary before? I wonder if we can stick to using physical metrics like shared memory usage instead of looking only at object store memory. Seems like we might still miss some memory usage if we only use object store memory?

Reply:
That is intentional. The shared mem in meminfo could be higher if the memory is released. If we look at meminfo we would be over-accounting. That is going to cause regression since today it is possible for the heap to occupy the memory from the object store

Reply:
maybe used_bytes - cached would be another option?

Reply:
Yea we already do that in GetLinuxMemoryBytes

 available_bytes = mem_free_bytes + cached_bytes + buffer_bytes;
 used_bytes = mem_total_bytes - available_bytes;

Reply:
we could add a different code path to read mem stat, but not sure why we need to add more code vs delete?